sdk/identity/Azure.Identity/src/MsalClientBase.cs

```diff
@@ -11,7 +12,7 @@ namespace Azure.Identity
     internal abstract class MsalClientBase<TClient>
         where TClient : IClientApplicationBase
     {
-        private readonly Lazy<Task> _ensureInitAsync;
+        private readonly AsyncLockWithValue<TClient> _clientAsyncLock;
```

Comment:
I'm a bit worried about refactoring classes which are unrelated to this fix so close to release.

Reply:
That is actually for the #13201. I didn't see that it is moved to the September release, so I can revert it.

Reply:
Reverted